### PR TITLE
[DependencyInjection] Fix tests of #18144

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

**Comment by HeahDude** [comment]
@nicolas-grekas Should be green now :)

**Comment by HeahDude** [comment]
If it goes in master we could use `finally`, right ?

**Comment by HeahDude** [comment]
Ok failure unrelated!

**Comment by HeahDude** [comment]
Apparently hhvm does not like `finally`... I revert that.

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @HeahDude.

**Comment by nicolas-grekas** [comment]
@weaverryan @dunglas I merged this one to make tests green again, please submit a PR if anything is wrong here (with test case :) )

File: src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php [hunk 1]
```diff
@@ -33,20 +32,7 @@ public function isFresh($timestamp)
...
-        try {
-            $reflectionClass = new \ReflectionClass($this->class);
-        } catch (\ReflectionException $e) {
-            // the class does not exist anymore!
-
-            return false;
-        }
-
-        return AutowirePass::createResourceForClass($reflectionClass);
+        return @filemtime($this->filePath) <= $timestamp;
```
Comment (weaverryan) [inline src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php]: @HeahDude thanks for fixing the test failure on #18144 - I didn't realize I had left an error!

But, this is wrong! This simplifies this resource down to be basically identical to the `SelfCheckingResourceInterface`.
The original last line *was* a mistake - I added a bug in my last commit - this was the mistake: https://github.com/symfony/symfony/pull/18144/commits/49271e86f999ad84bb4f8787b844973f7c22f4fc#diff-04e2904117929f6ffe784d5be3cb1103R49

So, this should be reverted, but and the last line corrected to be:

```php
return AutowirePass::createResourceForClass($reflectionClass); == $this;
```

Could you make a PR for that?

Comment (HeahDude) [inline src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php]: @weaverryan Yes, I tried to figure out what to do, I was not sure about that "easy" way :)

Thanks for the lead, I'm on it!

Comment (HeahDude) [inline src/Symfony/Component/DependencyInjection/Config/AutowireServiceResource.php]: @weaverryan Is there anyway to add a test for that ?

**Pull request closed without merging**